src/ui.form-select/USelect.vue

```diff
@@ -642,55 +148,76 @@ const isSelectedValueLabelVisible = computed(() => {
 const filteredOptions = computed(() => {
   const normalizedSearch = search.value.toLowerCase().trim() || "";
 
+  let selectedValues: (string | number)[] = [];
+
+  if (Array.isArray(props.modelValue)) {
+    selectedValues = props.modelValue.map((value) => {
+      if (typeof value === "object") {
+        return value[props.valueKey] as string | number;
+      }
+
+      return value;
+    });
+  } else if (props.modelValue) {
+    selectedValues =
+      typeof props.modelValue === "object"
+        ? [props.modelValue[props.valueKey]]
+        : [props.modelValue];
+  }
+
   let options = props.multiple
-    ? SelectService.removeSelectedValues(
-        props.options,
-        props.groupValueKey,
-        props.valueKey,
-        props.modelValue,
-      )
+    ? removeSelectedValues(props.options, selectedValues, props.valueKey, props.groupValueKey)
     : [...props.options];
 
   options = props.groupValueKey
-    ? filterAndFlat(options, normalizedSearch, props.labelKey)
-    : SelectService.filterOptions(options, normalizedSearch, props.labelKey);
+    ? filterGroups(
+        options,
+        normalizedSearch,
+        props.labelKey,
+        props.groupValueKey,
+        props.groupLabelKey,
+      )
+    : filterOptions(options, normalizedSearch, props.labelKey);
 
   return options.slice(0, props.optionsLimit || options.length);
 });
 
 const localValue = computed(() => {
   if (!props.multiple) {
-    return SelectService.getCurrentOption(
-      props.modelValue,
-      props.options,
-      props.groupValueKey,
-      props.valueKey,
-    );
+    const [singleValue] = Array.isArray(props.modelValue) ? props.modelValue : [props.modelValue];
+
+    return getCurrentOption(props.options, singleValue, props.valueKey, props.groupValueKey);
   }
 
-  return props.modelValue
-    ? props.modelValue.map((item) =>
-        SelectService.getCurrentOption(item, props.options, props.groupValueKey, props.valueKey),
+  return props.modelValue && Array.isArray(props.modelValue)
+    ? props.modelValue.map((value) =>
+        getCurrentOption(props.options, value, props.valueKey, props.groupValueKey),
       )
     : [];
 });
 
 const isLocalValue = computed(() => {
-  if (props.multiple) return Boolean(localValue.value.length);
+  const value = localValue.value;
 
-  return typeof localValue.value !== "number"
-    ? Boolean(localValue.value)
-    : Boolean(String(localValue.value));
+  if (Array.isArray(value)) {
+    return !!value?.length;
```

Comment:
`Boolean(value.length);`

It looks like a bunch of symbols now. And you don't need optional chaining operator here, array always have length property.

Reply:
!! is more readable and consistens in this case.

![image](https://github.com/user-attachments/assets/3ef01d33-46cd-4766-b7fc-ca2729b88ad2)